src/dev/flang/tools/Tool.java

```diff
@@ -249,15 +249,37 @@ else if (a.equals("-XjavaProf"))
       }
     else if (a.startsWith("-XjavaProf="))
       {
-        Profiler.start(a.substring(a.indexOf("=")+1));
+        var file = a.substring(a.indexOf("=")+1);
+        if (file.equals(""))
+          { fatal("Please provide a file name to option '-XjavaProf=<file>'."); }
+        else
+          { Profiler.start(); }
       }
-    else if (a.startsWith(Errors.MAX_ERROR_MESSAGES_OPTION) && a.startsWith(Errors.MAX_ERROR_MESSAGES_OPTION + "="))
+    else if (a.equals(Errors.MAX_ERROR_MESSAGES_OPTION))
       {
-        Errors.MAX_ERROR_MESSAGES = Integer.parseInt(a.substring(a.indexOf("=")+1));
+        Errors.MAX_ERROR_MESSAGES = -1;
       }
-    else if (a.startsWith(Errors.MAX_WARNING_MESSAGES_OPTION) && a.startsWith(Errors.MAX_WARNING_MESSAGES_OPTION + "="))
+    else if (a.startsWith(Errors.MAX_ERROR_MESSAGES_OPTION + "="))
       {
-        Errors.MAX_WARNING_MESSAGES = Integer.parseInt(a.substring(a.indexOf("=")+1));
+        try {
```

Comment:
why not use parsePositiveIntArg?

Reply:
to keep the possibility to set it to `=-1` for unlimited errors. Do you think it would be better to remove that?

Reply:
parsePositiveIntArg is misleading it also parses negative numbers. Please rename.